### PR TITLE
[Framework] this pr focus on adding smallest unit of GAS

### PR DESCRIPTION
**Pull request by Jim8y**
This pr focus on adding smallest unit of GAS,

it add satoshi, ksatoshi, msatoshi.

### CODE REVIEW

File: src/Neo.SmartContract.Framework/Neo.SmartContract.Framework.csproj [hunk 1]
```diff
@@ -14,6 +14,58 @@
...
+    </Compile>
+    <Compile Update="Contracts\Nep17Token.cs">
+      <Pack>true</Pack>
+      <PackagePath>src</PackagePath>
+    </Compile>
+    <Compile Update="ContractTypes\UInt160.cs">
+      <Pack>true</Pack>
+      <PackagePath>src</PackagePath>
+    </Compile>
+    <Compile Update="ContractTypes\UInt256.cs">
```
Comment (Jim8y) [inline src/Neo.SmartContract.Framework/Neo.SmartContract.Framework.csproj:49]:
```suggestion
```

File: src/Neo.SmartContract.Framework/Neo.SmartContract.Framework.csproj [hunk 2]
```diff
@@ -14,6 +14,25 @@
...
+      <PackagePath>src</PackagePath>
+    </Compile>
+    <Compile Update="Contracts\TokenContract.cs">
+      <Pack>true</Pack>
+      <PackagePath>src</PackagePath>
+    </Compile>
+    <Compile Update="ContractTypes\ContractParameterType.cs">
+      <Pack>true</Pack>
+      <PackagePath>src</PackagePath>
+    </Compile>
```
Comment (Jim8y) [inline src/Neo.SmartContract.Framework/Neo.SmartContract.Framework.csproj:35]:
```suggestion
```

File: tests/Neo.Compiler.CSharp.TestContracts/Contract1.cs [hunk 3]
```diff
@@ -1,10 +1,17 @@
...
     {
         private static string privateMethod()
         {
-            return "NEO3";
+            BigInteger a = 0;
+            var b = a * 1000;
+            var c = a * 1_000_000;
+            var d = a * 100_000_000;
+
+            return "NEO3" + b + c + d;
```
Comment (Jim8y) [inline tests/Neo.Compiler.CSharp.TestContracts/Contract1.cs:14]:
```suggestion
           return "NEO3";
```

File: tests/Neo.Compiler.CSharp.TestContracts/Contract1.cs [hunk 4]
```diff
@@ -1,10 +1,12 @@
+using System.Numerics;
+
 namespace Neo.Compiler.CSharp.UnitTests.TestClasses
 {
     public class Contract1 : SmartContract.Framework.SmartContract
     {
         private static string privateMethod()
         {
-            return "NEO3";
+           return "NEO3";
```
Comment (Jim8y) [inline tests/Neo.Compiler.CSharp.TestContracts/Contract1.cs:9]:
```suggestion
            return "NEO3";
```

File: tests/Neo.Compiler.CSharp.TestContracts/Contract1.cs [hunk 5]
```diff
@@ -27,6 +29,11 @@ public static byte[] testArgs1(byte a)
 
         public static object testArgs2(byte[] a)
         {
+            BigInteger aa = 0;
+            var b = aa * 1000;
+            var c = aa * 1_000_000;
+            var d = aa * 100_000_000;
+
```
Comment (Jim8y) [inline tests/Neo.Compiler.CSharp.TestContracts/Contract1.cs:36]:
```suggestion
```

File: tests/Neo.Compiler.CSharp.TestContracts/Contract1.cs [hunk 6]
```diff
@@ -1,3 +1,5 @@
+using System.Numerics;
+
```
Comment (Jim8y) [inline tests/Neo.Compiler.CSharp.TestContracts/Contract1.cs:2]:
```suggestion
```

**Review by cschuchardt88 — CHANGES_REQUESTED** [review]
Remove all the renames and create another `PR` or edit  `1st comment` and `title` to describe what this PR is suppose to do. Because as it sits; this `PR` isn't doing what it says it needs to do.

**Pull request closed without merging**